packages/api/lib/granules.js

```diff
@@ -62,10 +66,48 @@ function getGranuleProductVolume(granuleFiles = []) {
     .reduce((x, y) => x + y, 0);
 }
 
+/**
+ * Delete a Granule from Postgres and Dynamo, delete the Granule's
+ * Files from Postgres and S3
+ *
+ * @param {Object} params
+ * @param {Knex} params.knex - DB client
+ * @param {Object} params.dynamoGranule - Granule from DynamoDB
+ * @param {PostgresGranule} params.pgGranule - Granule from Postgres
+ * @param {FilePgModel} params.filePgModel - File Postgres model
+ * @param {GranulePgModel} params.granulePgModel - Granule Postgres model
+ * @param {Object} params.granuleModelClient - Granule Dynamo model
+ */
+const deleteGranuleAndFiles = async ({
+  knex,
+  dynamoGranule,
+  pgGranule,
+  filePgModel = new FilePgModel(),
+  granulePgModel = new GranulePgModel(),
+  granuleModelClient = new Granule(),
+}) => {
+  const files = await knex(filePgModel.tableName)
+    .where({ granule_cumulus_id: pgGranule.cumulus_id });
+
+  await knex.transaction(async (trx) => {
+    await pMap(
+      files,
+      (file) =>
+        knex.transaction(async (fileTrx) => {
+          await filePgModel.delete(fileTrx, { cumulus_id: file.cumulus_id });
+          await deleteS3Object(file.bucket, file.key);
```

Comment:
Nice work on this, but do we need to use the `FileUtils` to try and generate a `file.bucket` and `file.key`? Because the objects on the `granule.files` are not guaranteed to have those properties, though we might be able to generate them even if they don't.

This is what the old granule Dynamo delete logic was doing: https://github.com/nasa/cumulus/blob/master/packages/api/models/granules.js#L556

Reply:
Ohh sure thing. So I didn't do that originally because I came away thinking we would make `file` and `bucket` required on Files. That was after one of the conversations on Jennifer's ticket though but I think that's still very much up in the air. Using this logic is much safer.